src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ContextAdapterBase.cs

```diff
@@ -61,4 +73,117 @@ protected ContextAdapterBase(ICommandLineOptions commandLineOptions)
 
         return adapterSpecificTestCaseFilter;
     }
+
+    private void HandleFilter(ITestExecutionFilter? filter, string? filterFromRunsettings, string? filterFromCommandLineOption)
+    {
+        // No filters at all, we can return immediately as there is nothing to do.
+        if (filter is null or NopFilter && filterFromRunsettings is null && filterFromCommandLineOption is null)
+        {
+            return;
+        }
+
+        var filterBuilder = new StringBuilder();
+
+        AppendFilter(filterFromRunsettings, filterBuilder);
+        AppendFilter(filterFromCommandLineOption, filterBuilder);
+
+        // TODO: Handle TreeNodeFilter. Note that this is unlikely to be handled here.
+        // This code path tries to construct a VSTest compatible filter.
+        // This is probably not possible for TreeNodeFilter.
```

Comment:
I'd drop this comment. It won't help when implementing the `TreeNodeFilter` and if we don't find this line, it'll be there forever.